examples/plugin_class.py

```diff
@@ -0,0 +1,25 @@
+"""
+Display two image layers using the add_image API
+"""
+
+import sys
+from PyQt5.QtWidgets import QApplication
+
+from skimage import data
+from napari import Window, Viewer
+from napari.plugins import GaussianBlur
```

Comment:
Here's the vision I want for plugins:

```
from package import module  # contains Napari function annotations
import napari
napari.add_plugin(module)  # autopopulates menus with module functions
```

This works for functional modules, and maybe also for GUI modules. Without this, there is virtually zero chance of getting to the goal of napari recording pure Python. It's feasible, but it will never be Python that anyone would want to write.

Reply:
ah ok, this is nice to see - helps me understand. I definitely think we want to support this for the reasons you mention, and see how for large libraries with lots of functions that you want a menu bar to populate it makes sense. I'm just a little unsure how we will make the corresponding GUI elements for when those functions require additional parameters, but maybe this is doable and we just need to keep exploring.

If this becomes the main way to get custom code into napari then I might suggest that we "re-brand" our "plugin ecosystem" to a python "module ecosystem" and talk about `napari.add_module`

Reply:
> I'm just a little unsure how we will make the corresponding GUI elements for when those functions require additional parameters

The idea is to automatically determine the GUI elements from the input types. We had a discussion about this where @freeman-lab pointed us to such a framework for Javascript. 

> we "re-brand" our "plugin ecosystem" to a python "module ecosystem" and talk about `napari.add_module`

yes, that's exactly the right idea imho! Napari headless is just Python! =)